activemodel/lib/active_model/error.rb

```diff
@@ -69,9 +69,14 @@ def self.generate_message(attribute, type, base, options) # :nodoc:
 
       if base.class.respond_to?(:i18n_scope)
         i18n_scope = base.class.i18n_scope.to_s
+        unindexed_attribute = attribute.to_s.remove(/\[\d\]/)
+
         defaults = base.class.lookup_ancestors.flat_map do |klass|
-          [ :"#{i18n_scope}.errors.models.#{klass.model_name.i18n_key}.attributes.#{attribute}.#{type}",
-            :"#{i18n_scope}.errors.models.#{klass.model_name.i18n_key}.#{type}" ]
+          [
+            :"#{i18n_scope}.errors.models.#{klass.model_name.i18n_key}.attributes.#{attribute}.#{type}",
+            :"#{i18n_scope}.errors.models.#{klass.model_name.i18n_key}.attributes.#{unindexed_attribute}.#{type}",
```

Comment:
Do we need to add a new lookup? in `full_message` we just always remove the characters that matches `/\[\d\]/`.

Reply:
I was wondering about that myself - I'm not sure how safe it is to modify the existing lookup, because there could theoretically be somebody using the `\[\d\]` as part of their lookup. 

It is kind of ridiculous, and I imagine most people work around that instead, but removing that lookup would break the current expectation, right?

Reply:
It could, be I doubt it will. If people are using the index as part of their lookup they would have a huge translation fine. I'd just remove it, and if someone is relying on that behavior we put it back.

Reply:
Sounds good, I changed this to replace the existing lookup instead 👍